Lesson4/src/Homework4/Services/MilitaryPart.java

```diff
@@ -62,12 +61,47 @@ public boolean addRecruitToArray(Person person) {
     }
 
     public void printRecruitsFromArray() {
-        for (Person person : sizeMilitary) {
-            if (person != null) {
-                System.out.println(person.getName());
+        if (getFreePlacesFromArray() == size) {
+            System.out.println("Military part  " + unitNumber + " is free!");
+        } else {
+            if (sortSizeMilitary().length == 1) {
+                System.out.println("Military part\t" + unitNumber + "\thas a following soldier");
+                System.out.println((1) + "." + "\t" + sortSizeMilitary()[0].getName() + "\t" + sortSizeMilitary()[0].getSurName());
+            } else {
+                System.out.println("Military part\t" + unitNumber + "\thas the following soldiers");
+                for (int i = 0; i < sortSizeMilitary().length; i++) {
+                    System.out.println((i + 1) + "." + "\t" + sortSizeMilitary()[i].getName() + "\t" + sortSizeMilitary()[i].getSurName());
+                }
+
+            }
+            System.out.println("This military part has\t" + getFreePlacesFromArray() + "\tfree places");
+        }
+    }
+
+    private Person[] sortSizeMilitary() {
+        int count = 0;
+        for (int i = 0; i < sizeMilitary.length; i++) {
+            if (sizeMilitary[i] == null) {
+                count++;
             }
+        }
+        Person[] sortSizeMilitary = new Person[sizeMilitary.length - count];
+        for (int i = 0; i < sizeMilitary.length; i++) {
+            for (int j = 0; j < sortSizeMilitary.length; j++) {
+                if (sizeMilitary[i] != null && sortSizeMilitary[j] == null) {
+                    sortSizeMilitary[j] = sizeMilitary[i];
+                    break;
+                }
 
+            }
```

Comment:
this can be simplified to 
System.arraycopy(sizeMilitary, 0, sortSizeMilitary, 0, sizeMilitary.length - count);